docs/source/advanced/cmake.rst

```diff
@@ -0,0 +1,241 @@
+     Allow host-only contructs like assert in offload code in debug mode.
+
+.. _cpu-serial:
+
+CPU Serial
+----------
+
+ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE
+  .. code-block::
+
+     Enable the serial CPU back-end
+
+ALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB
+  .. code-block::
+
+     Kibibytes (1024B) of memory to allocate for block shared memory for backends
+     requiring static allocation
+
+.. _cpp-threads:
+
+C++ Threads
+-----------
+
+ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE
+  .. code-block::
+
+     Enable the threads CPU block thread back-end
+
+.. _boost-fiber:
+
+Boost Fiber
+-----------
+
+ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE
+  .. code-block::
+
+     Enable the fibers CPU block thread back-end
+
+.. _intel-tbb:
+
+Intel TBB
+---------
+
+ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE
+  .. code-block::
+
+     Enable the TBB CPU grid block back-end
+
+ALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB
+  .. code-block::
+
+     Kibibytes (1024B) of memory to allocate for block shared memory for backends
+     requiring static allocation
+
+.. _openmp2-grid-block:
+
+OpenMP 2 Grid Block
+-------------------
+
+ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE
+  .. code-block::
+
+     Enable the OpenMP 2.0 CPU grid block back-end
+
+ALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB
+  .. code-block::
+
+     Kibibytes (1024B) of memory to allocate for block shared memory for backends
+     requiring static allocation
+
+.. _openmp2-block-thread:
+
+OpenMP 2 Block thread
+---------------------
+
+ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE
+  .. code-block::
+
+     Enable the OpenMP 2.0 CPU block thread back-end
+
+.. _openmp5:
+
+OpenMP 5
+--------
+
+ALPAKA_ACC_ANY_BT_OMP5_ENABLE
+  .. code-block::
+
+     Enable the OpenMP 5.0 CPU block and block thread back-end
+
+
+ALPAKA_OFFLOAD_MAX_BLOCK_SIZE
+  .. code-block::
+
+     Maximum number threads per block to be suggested by any target offloading backends
+     ANY_BT_OMP5 and ANY_BT_OACC.
+
+.. _cuda:
+
+CUDA
+----
+
+ALPAKA_ACC_GPU_CUDA_ENABLE
+  .. code-block::
+
+     Enable the CUDA GPU back-end
+
+ALPAKA_ACC_GPU_CUDA_ONLY_MODE
+  .. code-block::
+
+     Only back-ends using CUDA can be enabled in this mode (This allows to mix
+     alpaka code with native CUDA code).
+
+
+ALPAKA_CUDA_ARCH
+  .. code-block::
+
+     GPU architecture: e.g. "35"
+
+ALPAKA_CUDA_COMPILER
+  .. code-block::
+
+     CUDA compiler: "nvcc" or "clang"
+
+ALPAKA_CUDA_FAST_MATH
+  .. code-block::
+
+     Enable fast-math
+
+ALPAKA_CUDA_FTZ
+  .. code-block::
+
+     Set flush to zero for GPU
+
+ALPAKA_CUDA_KEEP_FILES
+  .. code-block::
+
+     Keep all intermediate files that are generated during internal compilation steps
+     (folder: nvcc_tmp)
+
+ALPAKA_CUDA_NVCC_EXPT_EXTENDED_LAMBDA
+  .. code-block::
+
+     Enable experimental, extended host-device lambdas in NVCC
+
+ALPAKA_CUDA_NVCC_SEPARABLE_COMPILATION
+  .. code-block::
+
+     Enable separable compilation in NVCC
+
+https://developer.nvidia.com/blog/separate-compilation-linking-cuda-device-code/
```

Comment:
@psychocoderHPC Can you please verify the blog post.

Reply:
Not sure what I should verify but the blog post is showing how to use separate able compile

Reply:
I just want to know if the blog entry addresses the right topic and I haven't made a mistake. So, it is solved :-)